scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -989,6 +989,27 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     case _ => false
   }
 
+  def followedByToken[T <: Token: TokenInfo]: Boolean = {
+    def startBlock = token.is[LeftBrace] || token.is[Indentation.Indent]
+    def endBlock = token.is[RightBrace] || token.is[Indentation.Outdent]
+    @tailrec
+    def lookForToken(): Boolean = {
+      var braces = 0
+      if (braces == 0 && token.is[T]) true
+      else if (braces == 0 && (token.is[StopScanToken] || endBlock)) false
+      else if (token.is[EOF]) false
+      else {
+        if (startBlock) braces += 1
+        if (endBlock) braces -= 1
+        next()
+        lookForToken()
+      }
```

Comment:
`braces` should be a parameter in `lookForToken`.

Reply:
Good idea! Changed!